### DISCUSSION TITLE
URock: questions about raster output and grid size

### DISCUSSION THREAD

**Discussion by jlegewie** [UMEP-dev/UMEP#588] [Q&A] [answered]
Hi,

I have two questions about raster output  and grid size in URock:

1. How do the settings `RASTER_OUTPUT`, `HORIZONTAL_RESOLUTION` and `VERTICAL_RESOLUTION` work together? Looking at the code, they seem independent (if `RASTER_OUTPUT` is specified but `HORIZONTAL_RESOLUTION` and `VERTICAL_RESOLUTION` are not, they are set to the average x and y resolution of the raster). Does that mean that the initial calculation is done based on the resolution `HORIZONTAL_RESOLUTION` and `VERTICAL_RESOLUTION` and the results are projected to the template raster? Does it make any sense to to use a higher resolution for `HORIZONTAL_RESOLUTION` and `VERTICAL_RESOLUTION` (e.g. 2m) if the raster template has a lower resolution (e.g. 10m)?
3. How should I calculate the 3 dimensions that determine the overall cells? The documentation states that "very large grids (more than 30’000’000 3D grid cells) will not be possible to use". How do I calculate the number of cells? Let's say my template raster is 472 by 472 with a 2 meter resolution (222,784 cells) and `HORIZONTAL_RESOLUTION` and `VERTICAL_RESOLUTION` are set to 2 as well.

Thanks!

**Comment by j3r3m1** [comment, marked as answer]
Hi jlegewie,

1. You have understood and summarized well the code. I have just highlighted it below:

    - if RASTER_OUTPUT is specified but HORIZONTAL_RESOLUTION and VERTICAL_RESOLUTION are not, they are set to the average x and y resolution of the raster
    - if HORIZONTAL_RESOLUTION and VERTICAL_RESOLUTION are set, initial calculation is done based on the resolution HORIZONTAL_RESOLUTION and VERTICAL_RESOLUTION and the results are projected to the template raster

Concerning calculation at 2 m if you have a raster resolution of 10 m, you just have to keep in mind that the averaging will not take into account the wind direction. The absolute wind speed of all 2 m pixels in your 10 m pixels will be averaged. The results you will get if you directly calculate the wind at 10 m resolution will be the absolute value of the wind for a single pixel so it is not the same. 

Outside this difference of what you are interested in, the grid resolution you are using will affects your results differently depending on the size of the buildings and of the streets of your area of interest. As you can imagine, larger and simpler the buildings and the streets, larger can be the grid resolution. For the areas I have personally tested, there was almost no differences between 2 and 4 m resolution. But increasing above that size was (from what I remember) less fortunate. If you conduct a test on several areas, we would be interested by your results.

2. An accurate calculation cannot be made by hand since there is an extend of the zone of calculation along wind and cross wind (both horizontally and vertically), extend which is based on the size of the Röckle zones (which themselves depend on building size).
![image](https://github.com/UMEP-dev/UMEP/assets/13120405/8c019179-c48f-45df-b953-0157ee8a1247)

By default, the values for the extensions are 60, 40, and 20 m, respectively, for the along-wind, cross-wind, and vertical axis. You can get the level of magnitude of the zone considering only the "always added" extend but not the one coming from the zones. It would be a good first guess. Note that the bigger grid will be met for a wind coming from 45° from North, the cross wind and along wind size being calculated using all your raster but considered in the wind direction (sqrt(2)*472).
